File: Mobile/objects/viewmodels/Campaigns/StaffCampaignMobilesViewModel.ts
import { MobileCampaignStaffsViewModel } from "../CampaignStaff/MobileCampaignStaffsViewModel";
import { SchedulesViewModel } from "../Schedules/SchedulesViewModel";
import { IssuerViewModel } from "../Users/issuers/IssuerViewModel";

export interface StaffCampaignMobilesViewModel {
    id: number;
    code?: string;
    name?: string;
    description?: string;
    imageUrl?: string;
    format?: number;
    address?: string;
    startDate?: Date;
    endDate: Date;
    isRecurring?: boolean;
    status?: number;
    createdDate?: Date;
    updatedDate?: Date;
    statusName?: string;
    formatName?: string;
    sort?: string;
    issuers?: IssuerViewModel[];
    campaignStaffs?: MobileCampaignStaffsViewModel[];
    schedules?: SchedulesViewModel[];
    selectedSchedule?: SchedulesViewModel;
}

File: BE/Boek.Infrastructure/ViewModels/Campaigns/Mobile/StaffCampaignMobilesViewModel.cs
using System.Text.Json.Serialization;
using Boek.Infrastructure.Attributes;
using Boek.Infrastructure.ViewModels.CampaignStaffs;
using Boek.Infrastructure.ViewModels.Schedules;
using Boek.Infrastructure.ViewModels.Users.Issuers;

namespace Boek.Infrastructure.ViewModels.Campaigns.Mobile
{
    public class StaffCampaignMobilesViewModel
    {
        [Int]
        public int? Id { get; set; }
        [Guid]
        public Guid? Code { get; set; }

        [String]
        public string Name { get; set; }

        [String]
        public string Description { get; set; }

        [String]
        public string ImageUrl { get; set; }

        [Byte]
        public byte? Format { get; set; }

        [Byte]
        public string Address { get; set; }

        [StartDate]
        public DateTime? StartDate { get; set; }
        [EndDate]
        public DateTime? EndDate { get; set; }
        [Boolean]
        public bool? IsRecurring { get; set; }
        [Byte]
        public byte? Status { get; set; }

        [DateRange]
        public DateTime? CreatedDate { get; set; }

        [DateRange]
        public DateTime? UpdatedDate { get; set; }

        [String]
        public string StatusName { get; set; }

        [String]
        public string FormatName { get; set; }

        [Sort, JsonIgnore]
        public string Sort { get; set; }
        public List<IssuerViewModel> Issuers { get; set; }
        [JsonIgnore]
        public List<MobileCampaignStaffsViewModel> CampaignStaffs { get; set; }
        public List<SchedulesViewModel> Schedules { get; set; }
    }
}